exercises/practice/ledger/tests/ledger.cairo

```diff
@@ -0,0 +1,188 @@
+use ledger::{format_entries, Currency, Locale};
+
+#[test]
+fn empty_ledger() {
+    let currency = Currency::USD;
+    let locale = Locale::en_US;
+    let entries = array![];
+
+    let expect: Array<ByteArray> = array!["Date       | Description               | Change       "];
+    let result = format_entries(currency, locale, entries);
+
+    assert_eq!(result, expect);
+}
+
+#[test]
+#[ignore]
+fn one_entry() {
+    let currency = Currency::USD;
+    let locale = Locale::en_US;
+    let entries = array![("2015-01-01", "Buy present", "-1000")];
```

Comment:
I only used `byteArray` to make the entries consistent and mostly it will be easier to deal with positive and negative value, I will try and change it to signed integers and work around it

Reply:
@0xNeshi, I need help  trying to convert i32 data type to ByteArray, Do you have any suggestion that can help implement such?

Reply:
According to [canonical.json](https://github.com/exercism/problem-specifications/blob/2d26738ebb23f83ed6ab56d6d62566f3055f0f9a/exercises/ledger/canonical-data.json#L170-L176), the `amount_in_cents` is expected to be seperated with `,` or `.` based on the localed for value that are more than 5 digit for example `-123456` is expected to be `($1,234.56)` which means I have to be able to get into individual character to determine where the `,` or `.` is to be placed. using the `format!` macro won't give me the opprtunity to do that.

Reply:
```rust
    let y = -123456_i32;
    let cents = (y % 100) * -1; // In Cairo `%` includes negative values!
    let main = y / 100;
    let str = format!("{main}.{cents}");
    println!("{str}"); // prints "-1234.56"
    let str = format!("{main},{cents}");
    println!("{str}"); // prints "-1234,56"
```

Reply:
The expected data is supposed to be `-1, 234.56`  or  `-1.234,56` , the `,` and  `.`  that seperate the main value for readability are expected too

Reply:
You can use a similar approach to separate take 3-digit chunks out of a number by dividing it with `1000`, and concatenate with the appropriate delimiter.

We should follow canonical data and use signed integers. It should change the formatting logic, but it'll probably be even more straightforward.
Even if it's _less_ straightforward, the idea is _to solve the canonical exercise as it is_ using Cairo.

Reply:
You can take inspiration from [Python's solution](https://github.com/exercism/python/blob/main/exercises/practice/ledger/.meta/example.py)